### PR TITLE
Nh color strip v2

### PR DESCRIPTION
**Pull request by noahheller**

### CODE REVIEW

File: src/main/java/frc/robot/subsystems/LightStrip.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    private BlinkinPattern pattern = BlinkinPattern.WHITE;
+    private Map<Double, BlinkinPattern> delayedLightEvents = new TreeMap<>(Double::compareTo);
+    private Map<Callable<Boolean>, BlinkinPattern> predicateLightEvents = new HashMap<>();
+
+    public LightStrip(int port) {
+        this.colorSensorPort = new Spark(port);
+    }
+
+    @Override
+    public void periodic() {
```
Comment (goatfanboi23) [inline src/main/java/frc/robot/subsystems/LightStrip.java:29]: Todo make one map (easy)

File: src/main/java/frc/robot/commands/amp/ToggleAmp.java [hunk 2]
```diff
@@ -1,19 +1,21 @@
...
 
 public class ToggleAmp extends Command {
     private Amp amp;
     private Timer timeout = new Timer();
     private double speed;
-    private final TimeoutCounter timeoutCounter = new TimeoutCounter("ToggleAmp");
+    private final TimeoutCounter timeoutCounter;
 
-    public ToggleAmp(Amp amp) {
+    public ToggleAmp(Amp amp, LightStrip lightStrip) {
```
Comment (bengold22) [inline src/main/java/frc/robot/commands/amp/ToggleAmp.java:16]: Why bother to pass the lightstrip to these commands that don't need it?

File: src/main/java/frc/robot/subsystems/LightStrip.java [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+        colorSensorPort.set(pattern.getPwm());
+        SmartDashboard.putString("PATTERN",pattern.toString());
+        Logger.logDouble(baseLogName + "pwmSignal", pattern.getPwm(), Constants.ENABLE_LOGGING);
+        Logger.logString(baseLogName + "pwmName", pattern.toString(), Constants.ENABLE_LOGGING);
+    }
+
+    public BlinkinPattern getPattern() {
+        return pattern;
+    }
+    public void scheduleOnTrue(BooleanSupplier callable, BlinkinPattern pattern) {;
```
Comment (bengold22) [inline src/main/java/frc/robot/subsystems/LightStrip.java]: This even compile with the extra semi-colon?

Comment (noahheller) [reply src/main/java/frc/robot/subsystems/LightStrip.java]: Apparently it does compile. I don't know how it got there but I removed it

File: src/main/java/frc/robot/commands/amp/ToggleAmp.java (hunk 2 above)
Comment (noahheller) [reply src/main/java/frc/robot/commands/amp/ToggleAmp.java:16]: it is passed into the timeout counter in the command

**Review by batchen1 — APPROVED** [review]

**Pull request merged**